classes/Link.php

```diff
@@ -436,6 +437,12 @@ public function getCategoryLink(
             $category = $this->getCategoryObject($category, $idLang);
             $params['meta_title'] = Tools::str2url($category->getFieldByLang('meta_title'));
         }
+        if ($addPage) {
+            $page = (int) Tools::getValue('page');
+            if (false !== $page && $page > 1) {
+                $params['page'] = $page;
+            }
+        }
```

Comment:
Hmm, modifying the method signature for a single page of the front is not a good idea in my opinion.

It would be better in CategoryController::getCanonicalURL(), doesn't it?

Reply:
I hesitated at first, but it meant adding the param as a string with something like `$url .= "&page=$page"`
Whereas in the link method uses createUrl which is cleaner to generate urls
Besides the argument added is set to false by default so it won't cause any change of behaviour outside of this context
I can move the modification back to the CategoryController, I am just not fan of concatenating the param with a string

Reply:
@Quetzacoalt91 do you want me to change the behaviour or is it ok for you?

Reply:
Yep, the change I'd like is less clean. However you currently try to modify a widely used method for a specific use case, on a specific page. That's why I commented.

We can ask for another opinion, but if you end with a concatenation, this should be done with [http-build-query()](http://php.net/manual/en/function.http-build-query.php)

Reply:
ok, although I don't think it would have caused a problem since it used an extra parameter only used in this case I understand you concerns
I had the exact same ^^
I made the modifications, telle me if it is ok for you?